File: src_R/06_interval_proporsi.R
#Progam 6
interval_kepercayaan<-function(p_hat, n, alpha){
  if(p_hat < 0 || p_hat > 1){
    cat("Error: Proporsi harus bernilai antara 0 dan 1\n")
  } else{
    if(alpha == 0.10) z <- 1.645
    else if (alpha == 0.05) z <- 1.96
    else cat("Error: Alpha harus 0.05 atau 0.10\n")
    
    moe <- z * sqrt((p_hat * (1 - p_hat)) / n)
    lower <- p_hat - moe
    upper <- p_hat + moe
    
    cat("Interval Kepercayaan", (1 - alpha) * 100, "%\n")
    cat(lower, "< p <", upper, "\n")
  }
}
interval_kepercayaan(0.6, 100, 0.05)
interval_kepercayaan(0.4, 80, 0.05)
interval_kepercayaan(0.7, 120, 0.05)

File: src_python/06_interval_proporsi.py
from math import sqrt
def interval_kepercayaan(p_hat, n, alpha):
    # Validasi proporsi
    if p_hat < 0 or p_hat > 1:
        print("Error: Proporsi harus berada antara 0 dan 1.")
        return

    # Menentukan nilai z
    if alpha == 0.10:
        z = 1.645
    elif alpha == 0.05:
        z = 1.96
    else:
        print("Error: Alpha harus 0.10 atau 0.05.")
        return

    # Menghitung margin of error
    moe = sqrt((p_hat * (1 - p_hat)) / n)

    # Batas bawah dan atas
    lower = p_hat - (z*moe)
    upper = p_hat + (z*moe)
    print("interval kepercayaan :", (lower, upper))

interval_kepercayaan(0.6, 100, 0.05)
interval_kepercayaan(1, 100, 0.05)
interval_kepercayaan(0.6, 100, 0.01)
interval_kepercayaan(-1, 100, 0.05)
